subsys/nrf_rpc/nrf_rpc_uart.c

```diff
@@ -237,29 +283,61 @@ static void send_byte(const struct device *dev, uint8_t byte)
 static int send(const struct nrf_rpc_tr *transport, const uint8_t *data, size_t length)
 {
 	uint8_t crc[2];
+	uint16_t crc_val;
+	bool acked = true;
 	struct nrf_rpc_uart *uart_tr = transport->ctx;
 
+	k_mutex_lock(&uart_tr->tx_lock, K_FOREVER);
+
 	LOG_HEXDUMP_DBG(data, length, "Sending frame");
 
-	k_mutex_lock(&uart_tr->tx_lock, K_FOREVER);
+#if CONFIG_NRF_RPC_UART_RELIABLE
+	int attempts = 0;
```

Comment:
Should we keep declarations at the beginning (not sure how strict our coding guidelines is wrt that ;))?

Reply:
There are no any rules regarding this. I suggest keeping it `as is` to minimize the condition compilation code pollution.